source/selection_grid.c

```diff
@@ -57,12 +57,10 @@ void selection_grid_move_selection_vert(SelectionGrid* selection_grid, int direc
         if (new_row_size <= 0)
             return;
 
-        if (selection.x >= new_row_size)
-        {
-            // TODO: Maintain relative horizontal position
-            // Clip selection to row size
-            new_selection.x = new_row_size - 1;
-        }
+        int old_row_size = selection_grid->rows[selection.y].get_size();
+        // TODO: Maybe some of the fixed conversions here are unnecessary
+        new_selection.x = fx2int(fxmul(int2fx(selection.x), (int2fx(new_row_size) / old_row_size)));
```

Comment:
Here, if `old_row_size == 0`, then there will be a divide by 0 error. I think we should have a check for that being 0. Or (because I've been looking at branchless programming lately, lol). We could set the value to `1` if it's 0 without `if` statements with:

```c
old_row_size += (old_row_size == 0); 
```

We don't have to do that, but some sort of check is needed.